### DISCUSSION TITLE
[ERROR] GitHub Copilot could not connect to server. Extension activation failed: "Cannot read properties of null (reading 'uri')"

### DISCUSSION THREAD

**Discussion by sammcj** [community/community#43423] [Copilot Conversations] [answered]
### Select Topic Area

Bug

### Body

Over the past week I've seen a new bug with Copilot:

```
[ERROR] [default] [2023-01-05T22:51:33.002Z] GitHub Copilot could not connect to server. Extension activation failed: "Cannot read properties of null (reading 'uri')"
```

<img width="453" alt="SCR-20230106-dp2" src="https://user-images.githubusercontent.com/862951/210895490-4c2c6f58-a010-406b-968f-c636b372b824.png">

- It occurs every time I start VSCode.
- There's no way to permanently dismiss the dialogue.
- It seems to have no impact on the functionality of Copilot.

VSCode Version: 1.74.2
macOS Version: 13.1

**Comment by Akash1134** [comment]
Hi @sammcj , Are you connected to a network that's behind a VPN, or other technology that may be intercepting your connection to the server? Also, have you tried signing out of both the IDE and GitHub account,  sign back in to see if that helps.

**Reply by sammcj** [reply]
Howdy, no I'm not behind a VPN or Firewall that's MITMing TLS etc...

I logging out and back into GitHub yesterday.

Is there a way to get some better logging to see what the context of the error is?

**Comment by WilliamNurmi** [comment]
I'm having the same issue, I'm connected to internet without VPN.

**Comment by iamcjwilliams** [comment]
@sammcj @WilliamNurmi Also, please visit here to troubleshoot your issues: https://docs.github.com/en/copilot/troubleshooting-github-copilot

**Reply by UshakovMaksym** [reply]
This manual is not help with this type of problem, in logs we are seen the same text

**Comment by jamiepratt** [comment]
I'm having the same issue.

```
[ERROR] [default] [2023-01-11T17:49:35.723Z] GitHub Copilot could not connect to server. Extension activation failed: "Cannot read properties of null (reading 'uri')"
[ERROR] [default] [2023-01-11T17:49:40.193Z] GitHub Copilot could not connect to server. Extension activation failed: "Timed out waiting for authentication provider to register"
```
It does not seem to be a network problem:

```
# Reachability

- github.com: HTTP 200 - 
- copilot-proxy.githubusercontent.com: HTTP 200 - 
- default.exp-tas.com: HTTP 200 - 
```

**Reply by jamiepratt** [reply]
Disabling other extensions doesn't help.

**Reply by jamiepratt** [reply]
I installed and uninstalled visual studio code using "brew". This didn't help either.

**Reply by jamiepratt** [reply]
I'm not connected to a vpn.

**Comment by iamcjwilliams** [comment, marked as answer]
@jamiepratt Sorry for the trouble here, Could you provide a few more details on how this can be replicated?

**Reply by sammcj** [reply]
I'm really not sure sorry - it just seems to pop up randomly some times, I haven't seen it in the last 3 days though.

**Reply by jamiepratt** [reply]
Hi!

I am not getting the error now. It just started working without an error, I don't think I did anything to fix it.

This is going to be a difficult bug to track down it would seem.

Good luck!

Jamie

**Comment by UshakovMaksym** [comment]
I'm still have this warning, extension updates is not fix this bug

coPilot: Version | 1.70.8099

VS Code:
Version: 1.74.3 (user setup)
Commit: 97dec172d3256f8ca4bfb2143f3f76b503ca0534
Date: 2023-01-09T16:59:02.252Z
Electron: 19.1.8
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Windows_NT x64 10.0.19044
Sandboxed: No

**Comment by mukhammadazam** [comment]
error

**Comment by Mortezaegoe** [comment]
> #13131

**Comment by luni-moon** [comment]
I'm still getting this error (well, similar; output, `[ERROR] [default] [2023-02-01T21:46:08.213Z] GitHub Copilot could not connect to server. Extension activation failed: "Cannot read properties of undefined (reading 'registerActionRunner')"`), but it only happens on a local VSCode, on a specific machine. It does not do this on a device using Codespaces (last I checked), (both machines) with settings sync enabled. I am running the most recent Windows 10 build and have tried all other "fixes" in this thread, as well as not ever being connected to a VPN or similar, but still can't put a muzzle on the error. Any more help would be appreciated! Thanks!

**Comment by MichaelReaNZ** [comment]
I'm getting it randomly too.

**Comment by tonyt-dev** [comment]
This error does not go away and I'm paying for copilot, it works in VS Code Insiders edition if that helps. Tried pretty much everything short of formatting the machine

**Reply by JustinHaut** [reply]
Go to the profiles section in VSCode and make sure Github Authentication is enabled.

<img width="1293" alt="Screenshot 2023-03-01 at 7 44 06 PM" src="https://user-images.githubusercontent.com/19810494/222309536-fd3f0fd0-7953-476a-8d90-1b7d1cb123b7.png">

**Reply by airampg** [reply]
> Go to the profiles section in VSCode and make sure Github Authentication is enabled.


Yes! That helped me solve this issue. Thanks a lot! I had disabled it and didn't know how to enable it back.

**Reply by jaf7** [reply]
This was my issue.

**Comment by luni-moon** [comment]
This didn’t solve my issue, it works now for no reason, but I never
disabled that.

On Mon, Mar 6, 2023 at 08:26, Airam Perez Guillen ***@***.***>
wrote:

> Go to the profiles section in VSCode and make sure Github Authentication
> is enabled.
> [image: Screenshot 2023-03-01 at 7 44 06 PM]
> <https://user-images.githubusercontent.com/19810494/222309536-fd3f0fd0-7953-476a-8d90-1b7d1cb123b7.png>
>
> Yes! That helped me solve this issue. Thanks a lot! I had disabled it and
> didn't know how to enable it back.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/community/discussions/43423#discussioncomment-5215900>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANI5RBJ33GRNLVAITHUIW2LW2XQXVANCNFSM6AAAAAATSO7ZS4>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>
-- 
Best Wishes,
Luke Batema (JavaScript, C#, DiscordJS, Python, and Front End Developer,
Creator of cubelang)
